pkg/reconcile/reconcilers.go

```diff
@@ -178,6 +185,59 @@ func (r *defaultReconciler) ReconcileSchema(ctx context.Context, req ReaperReque
 	}
 }
 
+func (r *defaultReconciler) checkForCassandraDatacenterReadiness(ctx context.Context, req ReaperRequest) error {
+	reaper := req.Reaper
+	cassandra := *reaper.Spec.ServerConfig.CassandraBackend
+	cassdc := &cassdcapi.CassandraDatacenter{}
+
+	fetchStatus := func() error {
+		cassdcKey := types.NamespacedName{Namespace: reaper.Namespace, Name: cassandra.DatacenterName()}
+		err := r.Client.Get(ctx, cassdcKey, cassdc)
+		if runtime.IsNotRegisteredError(err) {
```

Comment:
You should not need to check for a not registered error. We register the CassandraDatacenter schema at start up [here](https://github.com/thelastpickle/reaper-operator/blob/master/main.go#L49).

Reply:
Without the check, the tests will fail.

Reply:
Although I guess these are not missing on purpose from the suite_test.go .. I'll try fixing it.

Reply:
Yeah, the CassandraDatacenter scheme  needs to be added in the integration test. Sorry, I was only thinking before about the e2e test.